### PR TITLE
#120: API-core-policies-TeamHandleConcept transition from md to rst

### PR DESCRIPTION
**Pull request by antoinemeyer5**
* API/core/policies/TeamHandleConcept transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------: 
image | image
image | --- 
image | image
image | image
image | image

### CODE REVIEW

**Comment by fnrizzi** [comment]
something seems off here, the rendering of the new page has lots of things missing

**Comment by antoinemeyer5** [comment]
> something seems off here, the rendering of the new page has lots of things missing

Yes, the result is really different.

Pierre had also done this here #237 and I was inspired by it. The idea is that rather than separating the function (in synopsis) and its explanations (in Public Class Members), we move the explanations.

I move the explanations of the class functions that were previously in plain text into the class synopsis section with sphinx syntax.

This is a choice but may be wrong or not the result you want. I'm waiting for your feedback on my current proposal + explanation but if it doesn't fit I'll go back to the usual conversion method.

I hope not to waste too much of your time. Thanks in advance for your feedback @fnrizzi !

**Comment by antoinemeyer5** [comment]
Hello, I made the changes with `kokkosinlinefunction`.
I finish the screenshots when #256 will be merged.

**Comment by fnrizzi** [comment]
yes because we cannot do anything until that PR is  merged

**Review by ajpowelsnl — APPROVED** [review]
This generally looks good, but it's a bit confusing in certain places.  In future iterations, we can polish the content.

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (ajpowelsnl) [inline docs/source/API/core/policies/TeamHandleConcept.rst:37]: In comparing the "Before" and "After," is the key difference showing _only_ the public API in the .rst version?  Assuming this is the case, I like it : )

That being said, I do think the section headings (e.g., Index Queries, etc.) should be used.

**Review by ajpowelsnl — CHANGES_REQUESTED** [review]
Please use the section headings from the .md files

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (antoinemeyer5) [reply docs/source/API/core/policies/TeamHandleConcept.rst:117]: Hello @ajpowelsnl,

The reason is simple: we often need to write`KOKKOS_INLINE_FUNCTION something`.
For example: `KOKKOS_INLINE_FUNCTION int team_size() const noexcept`.

In the markdown code this was translated into a block of c++ code: ```c++`` where we had to write in hardcopy the `KKOKKOS_INLINE_FUNCTION` "symbol".

Comment (antoinemeyer5) [reply docs/source/API/core/policies/TeamHandleConcept.rst:117]: For the second question and the PR of @nmm0 , I don't know how to answer you. You have to ask @fnrizzi who has a more global vision than me.

I hope I have answered your questions well and justified why this specialization exists.

Kind regards

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (antoinemeyer5) [reply docs/source/API/core/policies/TeamHandleConcept.rst:37]: We discussed with @fnrizzi and yes I am now using section headings.

The end result will be like the original .md file but with the specifications for functions starting with `KOKKOS_INLINE_FUNCTION`.

I haven't updated the screenshots because you have to validate the request #256 first.

=> The section titles are back, no worries.

Comment (antoinemeyer5) [reply docs/source/API/core/policies/TeamHandleConcept.rst:37]: To answer your question, yes, the purpose is to show the visual difference in order to already validate if the new visual is correct.

Then the code can be inspected by the Reviewers. As the visual is the most important for the future users and for the Reviewers who validate or not the changes, it is much more convenient for you to already show the final result.

**Comment by antoinemeyer5** [comment]
Hello @fnrizzi, @ajpowelsnl & @pierrepebay, I made the changes with kokkosinlinefunction and updated the screenshots. I have taken the original structure. Kind regards.

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:120]: indentation needs to be fixed

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see coments

**Comment by antoinemeyer5** [comment]
> see coments

Corrected, see screenshots @fnrizzi

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:163]: use .. cppkokkos:kokkosinlinefunction::

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:174]: use .. cppkokkos:kokkosinlinefunction::

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:186]: use .. cppkokkos:kokkosinlinefunction::

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:195]: use .. cppkokkos:kokkosinlinefunction::

**Review by fnrizzi — CHANGES_REQUESTED** [review]
see comments

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:76]: should use 
```
.. cpp:type:: execution_space
```

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [inline docs/source/API/core/policies/TeamHandleConcept.rst:78]: similarly to above

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (antoinemeyer5) [reply docs/source/API/core/policies/TeamHandleConcept.rst:89]: Source code: https://github.com/kokkos/kokkos/blob/master/core/src/Kokkos_HPX.hpp

TeamHandleConcept is called HPXTeamMember (from line 650).

I understand little by little how everything fits together. But for me there are no constructors or destructors.. can you help me with this problem... I may be looking in the wrong place.

Comment (fnrizzi) [reply docs/source/API/core/policies/TeamHandleConcept.rst:89]: yes sure i will take a look!

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (dalg24) [inline docs/source/API/core/policies/TeamHandleConcept.rst:143]: Not rendered properly

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (dalg24) [inline docs/source/API/core/policies/TeamHandleConcept.rst:145]: Same comment

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [reply docs/source/API/core/policies/TeamHandleConcept.rst]: thanks for catching this, solved!

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [reply docs/source/API/core/policies/TeamHandleConcept.rst]: thanks for catching this, solved!

**Review by fnrizzi — APPROVED** [review]

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (ajpowelsnl) [inline docs/source/API/core/policies/TeamHandleConcept.rst:10]: Which following "API requirements?" : )

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (ajpowelsnl) [inline docs/source/API/core/policies/TeamHandleConcept.rst:16]: There should be no comment on things that are not part of the public API.  Please remove non-public API material.  Perhaps you could simply delete line 13 -- it's not very informative, in any case.

**Review by ajpowelsnl — APPROVED** [review]
As mentioned earlier, please preserve the headings used in the markdown files and make sure the .md vs. .rst files show similar structure and content.

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [reply docs/source/API/core/policies/TeamHandleConcept.rst:10]: resolved

File: docs/source/API/core/policies/TeamHandleConcept.rst
Comment (fnrizzi) [reply docs/source/API/core/policies/TeamHandleConcept.rst:16]: I think we should keep this because it stresses and reminds users that this is just the concept. Repetita iuvant, especially for things as delicate as these.

**Pull request merged**
